Tests/Integration/RedisFixture.cs

```diff
@@ -124,17 +125,32 @@ public async Task InitializeAsync()
             await StartRedisContainer();
 
             var redisConnectionString = $"localhost:{RedisPort}";
-            try
+            var isRedisReady = false;
+            var maxNumberOfRetries = 10;
+            var tryRun = 0;
+            while (!isRedisReady && tryRun < maxNumberOfRetries)
             {
-                this.Redis = await ConnectionMultiplexer.ConnectAsync(redisConnectionString);
-                Database = this.Redis.GetDatabase();
+                try
+                {
+                    Redis = await ConnectionMultiplexer.ConnectAsync(redisConnectionString);
+                    Database = this.Redis.GetDatabase();
+                    isRedisReady = true;
+                }
+                catch (Exception ex)
```

Comment:
nit: is there a way to catch a more specific exception? ideally we should retry only if we could not connect, but not if we could not get the database, no?

Reply:
Yes I was lazy here, let me try to make this better